src/arm/windows/init_by_logical_sys_info.c

```diff
@@ -0,0 +1,842 @@
+	{
+		if (info->Relationship != info_type) {
+			cpuinfo_log_warning(
+				"unexpected processor info type (%"PRIu32") for processor information",
+				(uint32_t) info->Relationship);
+			continue;
+		}
+
+		const uint32_t info_id = nr_of_structs++;
+		uint32_t group_count = 0;
+
+		switch(info_type) {
+			case RelationProcessorPackage:
+				parse_relation_processor_info(
+					processors,
+					number_of_processors,
+					global_proc_index_per_group,
+					info,
+					info_id,
+					cores,
+					chip_info);
+			break;
+			case RelationProcessorCore:
+				parse_relation_processor_info(
+					processors,
+					number_of_processors,
+					global_proc_index_per_group,
+					info,
+					info_id,
+					cores,
+					chip_info);
+			break;
+			case RelationCache:
+				parse_relation_cache_info(
+					processors,
+					number_of_processors,
+					caches,
+					numbers_of_caches,
+					global_proc_index_per_group,
+					info,
+					info_id);
+			break;
+			default:
+				cpuinfo_log_error(
+					"unexpected processor info type (%"PRIu32") for processor information",
+					(uint32_t) info->Relationship);
+			break;
+		}
+		
+	}
+	/* 5. Release dynamically allocated info structure. */
+	HeapFree(heap, 0, infos);
+	return nr_of_structs;
+}
+
+static void parse_relation_processor_info(
+	struct cpuinfo_processor* processors,
+	uint32_t nr_of_processors,
+	const uint32_t* global_proc_index_per_group,
+	PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX info,
+	const uint32_t info_id,
+	struct cpuinfo_core* cores,
+	const struct woa_chip_info *chip_info)
+{
+	for (uint32_t i = 0; i < info->Processor.GroupCount; i++) {
+		const uint32_t group_id = info->Processor.GroupMask[i].Group;
+		/* Bitmask representing processors in this group belonging to this package */
+		KAFFINITY group_processors_mask = info->Processor.GroupMask[i].Mask;
+		while (group_processors_mask != 0) {
+			const uint32_t processor_id_in_group =
+				low_index_from_kaffinity(group_processors_mask);
+			const uint32_t processor_global_index =
+				global_proc_index_per_group[group_id] + processor_id_in_group;
+
+			if(processor_global_index >= nr_of_processors) {
+				cpuinfo_log_error("unexpected processor index %"PRIu32"",
+					processor_global_index);
+			}
+
+			switch(info->Relationship) {
+				case RelationProcessorPackage:
+					store_package_info_per_processor(
+						processors, processor_global_index, info_id,
+						group_id, processor_id_in_group,
+						chip_info);
+				break;
+				case RelationProcessorCore:
+					store_core_info_per_processor(
+						processors, processor_global_index,
+						info_id, info,
+						cores, chip_info);
+				break;
+				default:
+					cpuinfo_log_error(
+						"unexpected processor info type (%"PRIu32") for processor information",
+						(uint32_t) info->Relationship);
+				break;
+			}
+			/* Reset the lowest bit in affinity mask */
+			group_processors_mask &= (group_processors_mask - 1);

```

Comment:
This resets the most significant bits, not the lowest bit.

Reply:
I've inherited this code from x86 init: https://github.com/pytorch/cpuinfo/blob/master/src/x86/windows/init.c#L229
I mean it's absolutely not an excuse, I'll check that again.

Reply:
I think it really clears all bits, that are lower than the least set bit.

`ff	11111111
fe	11111110
fc	11111100
f8	11111000
f0	11110000
e0	11100000
c0	11000000
80	10000000
0	00000000`

`0x10
0x0

0x20
0x0

0x40
0x0

0x80
0x0`